runtime:compiler:feat: unify node-based makeprgs

Konfekt opened the pull request:
Follow-up to closed https://github.com/vim/vim/pull/18755 
Now defaults to `npx` as suggested by @romainl and @dkearns ;  let the user optionally change it by `b/g:node_makeprg` and documented where documentation existed; @dkearns 's `b/g:tsc_makeprg` kept for backwards compatibility; would that be acceptable, @djmoch  ?

Konfekt committed:
runtime:compiler:feat: unify node-based makeprgs

Konfekt committed:
feat: add per-compiler makeprg checks and prefer them over node_makeprg

Konfekt committed:
refactor: shorten makeprg setup with ternary in compilers

Konfekt committed:
use javascript_node_makeprg instead of node_makeprg as suggested by @dkearns

Konfekt committed:
fix: mixed indent

djmoch commented:
> would that be acceptable, @djmoch ?

If you're going to make changes, please remove me as the maintainer. I am not really using these anymore and don't want to stand in the way of anyone else making improvements. Perhaps you should adopt them yourself, @Konfekt.

Konfekt committed:
remove @djmoch as maintainer

Konfekt commented:
@djmoch When you say changes, do you mean any change to the file or in substance? Replacing

```vim
get(g:, 'csslint_makeprg', '')
```

to

```vim
get(g:, 'csslint_makeprg', 'csslint')
```

will fully restore the previous behavior of, for example, csslint.
Same can be done for the other file.

chrisbra commented on runtime/compiler/biome.vim:
```diff
@@ -2,15 +2,21 @@
+" CompilerSet makeprg=npx\ biome
+exe 'CompilerSet makeprg=' .. escape(
+                      \ (!empty(get(b:, 'biome_makeprg', get(g:, 'biome_makeprg', ''))) ?
+                      \   get(b:, 'biome_makeprg', get(g:, 'biome_makeprg', '')) :
```
Biome is quite new, let's rather keep it simple and jut use the `javascript_node_makeprg` instead and get rid of the whole `biome_makeprg` option instead. This simplifies this whole thing here.

chrisbra commented on runtime/compiler/eslint.vim:
```diff
@@ -2,11 +2,19 @@
+" CompilerSet makeprg=npx\ eslint
+exe 'CompilerSet makeprg=' .. escape(
+                        \ (!empty(get(b:, 'eslint_makeprg', get(g:, 'eslint_makeprg', ''))) ?
+                        \   get(b:, 'eslint_makeprg', get(g:, 'eslint_makeprg', '')) :
```
similar here please

Konfekt committed:
use tabs for erroformat as kindly pointed out by @chrisbra

applies to most of @dkearn's compiler files; issue as always some git
setting suppressing white space changes

Konfekt commented:
> Thanks, but I am not so happy with introducing a whole batch of new option variables and in addition use those `javascript_node_` variables as fallback. Can't we stick to one single way to do it? This is quite hard to understand and maintain as it is now.

See https://github.com/vim/vim/pull/18798#issuecomment-3566041100 regarding variable pollution; one could guard them behind their main file types as well, though, say, biome treats many. 

Having an executable in PATH to apply formatting independent of the repo one is in still has use cases, say scratch files, docs, ...

From what I saw, `npx` in HOME will install node_modules in HOME, rather than globally `npm install -g`, so I don't know how to offer that other than giving the user the option to do so.

Konfekt commented:
> I don't have a good short term solution but these variables are really starting to make a mess of the global namespace

Most natural would be simply inverting the order, `b:makeprg_(params_)tsc`, so that suffixed `makeprg` vars are used for compiler settings. Adding backward compatible mappings (if ever deemed worth it) will not be too much effort.

Of course, one could also think even larger, and agree on a `g:vimrc` prefix or similar, so that `g:vimrc_makeprg` indeed should be rather unique

chrisbra commented:
> From what I saw, `npx` in HOME will install node_modules in HOME, rather than globally `npm install -g`, so I don't know how to offer that other than giving the user the option to do so.

Sorry, I don't follow. Why can't you use consistently `npx` in that case then?

chrisbra commented:
> I would be happy enough just to use npx for any of these NPM packaged tools and call it a day

I tend to agree. Thanks

Konfekt commented:
> I strongly dislike the use of config variables for anything that can be trivially managed with the "after-directory" 

A point of contention could be what counts as trivial here; what's trivial for experienced maintainers of Vim runtime files might be non-obvious for a user working on the Vim documentation level; less experienced users might prefer to set customizing options in the Vim help, instead of forking runtime files into the user's directory and editing suitably (which, as they might go out of sync, could also be a preferred option for those having that expertise, but that was already mentioned above).

Another advantage to using a variable to tweak the makeprg options is that this is more easily done at a  (node, python, ..) project local vimrc , with no need to add in it additional (after) runtime paths
